radar_libraries.csv

```diff
@@ -1,59 +0,0 @@
-name,ring,quadrant,isNew,description
-"lodash",adopt,NPM Packages,false,"<a href='https://github.com/lodash/lodash'>Lodash</a> is a multi-purpose utility library for JavaScript."
-"moment",adopt,NPM Packages,false,"<a href='https://github.com/moment/moment'>Moment</a> is a library for parsing, validating, manipulating, and formatting dates."
-"numeral",adopt,NPM Packages,false,"<a href='http://numeraljs.com/'>Numeral</a> is a Javascript library for formatting and manipulating numbers."
-"highcharts",adopt,NPM Packages,false,"<a href='https://www.highcharts.com/'>Highcharts</a> is a commercial Javascript charting library, used heavily in SQL Monitor."
-"underscore",endure,NPM Packages,false,"<a href='https://github.com/jashkenas/underscore'>Underscore</a> is a multi-purpose utility library for Javascript. We prefer lodash."
-"day.js",explore,NPM Packages,false,"<a href='https://github.com/iamkun/dayjs'>Day.js</a> is a trivially-small date library, and an alternative to moment.js. Being explored by CoreDev."
-"Monaco Editor",explore,NPM Packages,true,"<a href='https://microsoft.github.io/monaco-editor/'>Monaco</a> is the editor behind VS Code, available as a library for applications. It provides a code editor with completion/highlighting and a diff viewer. Currently being explored by Team Orca for editing/viewing migration scripts."
-"jscookie",adopt,NPM Packages,false,"<a href='https://github.com/js-cookie/js-cookie'>Jscookie</a> is a lightweight JavaScript API for handling browser cookies."
-"fetch",adopt,NPM Packages,false,"<a href='https://github.com/github/fetch'>Fetch</a> enables use of the <a href='https://fetch.spec.whatwg.org/'>fetch standard</a> for request/response inside a web browser."
-"axios",endure,NPM Packages,false,"<a href='https://github.com/axios/axios'>Axios</a> is a promise-based HTTP client. We only use this when there is a requirement to support IE9/10."
-"System.Commandline",adopt,Public NuGet,true,"<A href='https://github.com/dotnet/command-line-api'>System.Commandline</a> is a Microsoft library for command line parsing. It's well supported and our default choice for new projects."
```

Comment:
I'm sad about seeing `radar_libraries.csv` being removed.
I really think that there is a value in standardizing usage of some of the lower-level libraries too.

Reply:
👍 I agree there's some value (e.g. `System.Commandline` ), but there's also stuff of dubious value (does guidance on using `jscookie` add anything?).

Do you have any suggestions around heuristics to make it consistently useful?

Reply:
My proposal would be to list libraries that meet one of these criteria:
- there is a high chance that we need them for any new (.NET) project (eg. `NUnit` or `NSubstitute`)
- there is a common need for functionality that it provides and there are multiple options available (e.g. `System.Commandline`)

Also, list only ones that we recommend, perhaps only mentioning that we prefer them over alternatives (for discoverability) - e.g. "`NSubstitute` - mocking library - prefer it over `Moq` or `Rhino.Mocks`".

The immediate problem that I had to deal with was that for SQL Census (which was a greenfield product obviously) `Rhino.Mocks` was used as a mocking library. Once it was merged with the SQL Monitor codebase, [we had to deal with it](https://github.com/red-gate/sqlmonitor/pull/6339) at some point, as having it prevented the move to .NET Core.

I hope that having a reference like this would prevent the team from using `Rhino.Mocks` in the first place.

Reply:
This was moved to https://info.red-gate.com/display/PD/Recommended+libraries. Resolving.